translation/message_format.rst

```diff
@@ -0,0 +1,436 @@
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        finish_place: >
+            You finished {place, selectordinal,
+                one {#st}
+                two {#nd}
+                few {#rd}
+                other {#th}
+            }!
+
+    .. code-block:: xml
+
+        <!-- translations/messages.en.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="finish_place">
+                        <source>finish_place</source>
+                        <target>{apples, plural, =0 {There are no apples :(} one {There is one apple...} other {There are # apples!}}</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'finish_place' => 'You finished {place, selectordinal,
+                one {#st}
+                two {#nd}
+                few {#rd}
+                other {#th}
+            }!',
+        ];
+
+.. code-block:: php
+
+    // ...
+
+    // prints "You finished 1st!"
+    echo $translator->trans('finish_place', ['place' => 1]);
+
+    // prints "You finished 9th!"
+    echo $translator->trans('finish_place', ['place' => 9]);
+
+    // prints "You finished 23rd!"
+    echo $translator->trans('finish_place', ['place' => 23]);
+
+The possible cases for this are also shown in Unicode's `Language Plural Rules`_ document.
+
+Date and Time
+~~~~~~~~~~~~~
+
+The date and time function allows you to format dates in the target locale
+using the :phpclass:`IntlDateFormatter`:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # translations/messages.en.yaml
+        published_at: 'Published at {time, date} - {time, time, short}'
+
+    .. code-block:: xml
+
+        <!-- translations/messages.en.xlf -->
+        <?xml version="1.0"?>
+        <xliff version="1.2" xmlns="urn:oasis:names:tc:xliff:document:1.2">
+            <file source-language="en" datatype="plaintext" original="file.ext">
+                <body>
+                    <trans-unit id="published_at">
+                        <source>published_at</source>
+                        <target>Published at {time, date} - {time, time, short}</target>
+                    </trans-unit>
+                </body>
+            </file>
+        </xliff>
+
+    .. code-block:: php
+
+        // translations/messages.en.php
+        return [
+            'published_at' => 'Published at {time, date} - {time, time, short}',
+        ];
+
+.. code-block:: php
+
+    // ...
+
+    // prints "Published at Jan 25, 2019 - 11:30 AM"
+    echo $translator->trans('published_at', ['time' => new \DateTime('2019-01-25 11:30:00')]);
```

Comment:
Using `time` as the variable name is very confusing later when using it in the translation; example: `{time, time, short}` <- which is the variable and which is the function name? We could use a long variable name instead: `publication_date`